openfeature/multiprovider/isolation.go

```diff
@@ -0,0 +1,340 @@
+package multiprovider
+
+import (
+	"context"
+	"fmt"
+	of "github.com/open-feature/go-sdk/openfeature"
+	"sync"
+)
+
+type (
+	// HookIsolator used as a wrapper around a provider that prevents context changes from leaking between providers
+	// during evaluation
+	HookIsolator struct {
```

Comment:
You can embed the `openfeature.UnimplementedHook` in this struct so that you don't have to define empty `After`, `Error`, and `Finally` methods.